### ISSUE TITLE
limit is added to query even when none specified

### ISSUE THREAD

**Issue by rudmer** [dbeaver/dbeaver#6717]
#### System information: 
- Operating system (distribution) and version: **Ubuntu Linux 19.04**
- DBeaver version: **Version 6.2.0.201909081341**
- Additional extensions: _none_

#### Connection specification:
- Database name and version: **Vertica 9.2.0-0**
- Driver name: **vertica-jdbc-9.2.0-0.jar**
- Do you use tunnels or proxies (SSH, SOCKS, etc)? **no** _local database_

#### Describe the problem you're observing:
previously I could execute this query without any problems, after DBeaver upgrade it does not work anymore (database is the same as before):
```sql
select set_config_parameter('AWSRegion','us-west-2');
```
when executing this query ``LIMIT <resultset fetch-size>`` is added to the query:
![image](https://user-images.githubusercontent.com/8223430/64671424-b3382800-d468-11e9-9059-81bd1618bb81.png)

#### Steps to reproduce, if exist:
- execute above query
- set _result fetch-size_ to nothing (remove value)
- execute query again
- query fails again and _result fetch-size_ is same as before...

#### Include any warning/errors/backtraces from the logs

```
org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [5297] [0A000]: [Vertica][VJDBC](5297) ERROR: Unsupported use of LIMIT/OFFSET clause
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:134)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:473)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$0(SQLQueryJob.java:412)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:148)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:405)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:865)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:2833)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:98)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:148)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:96)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$17.run(ResultSetViewer.java:3356)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:103)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.sql.SQLFeatureNotSupportedException: [Vertica][VJDBC](5297) ERROR: Unsupported use of LIMIT/OFFSET clause
	at com.vertica.util.ServerErrorData.buildException(Unknown Source)
	at com.vertica.dataengine.VQueryExecutor.executeSimpleProtocol(Unknown Source)
	at com.vertica.dataengine.VQueryExecutor.execute(Unknown Source)
	at com.vertica.jdbc.common.SStatement.executeNoParams(Unknown Source)
	at com.vertica.jdbc.common.SStatement.execute(Unknown Source)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:338)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	... 12 more
Caused by: com.vertica.support.exceptions.FeatureNotSupportedException: [Vertica][VJDBC](5297) ERROR: Unsupported use of LIMIT/OFFSET clause
	... 19 more
```

**Comment by hannahshachar** [comment]
Yes, I have the same problem and I am unable to run select close_session() in Vertica due to this.

**Comment by oBoMBaYo** [comment]
Yes, we have the same problem and a lot of command unable to run such as select export_objects('', '<schema_name.table_name>');, select close_session();, select get_node_dependencies();, etc....

**Comment by rudmer** [comment]
looked somewhat further and I already had the setting `use SQL to limit fetch size` disabled, so it seems that that setting is not working anymore...

![image](https://user-images.githubusercontent.com/8223430/64692746-1ee4ba00-d496-11e9-8438-e9d4f46b8a13.png)

**Comment by Timuchin** [comment]
Same problem

**Comment by serge-rider** [comment]
I can't reproduce this.
It fails with this error only if "Use SQL to limit fetch size" is enabled (DBeaver appends explicit LIMIT to add SELECT queries in this case).
But with this option disabled everything works ok.

@rudmer Please check that this option is disabled for your particular Vertica connection:
![image](https://user-images.githubusercontent.com/6398845/64992923-8eff9f80-d8dd-11e9-9ced-f863fadf7555.png)

**Comment by serge-rider** [comment]
Generally we could try to make this SQL modifications smarter. For example do not add if there is no FROM clause. 
Although I'm not sure that this is good solution (is it possible that some function called in simple SELECT will return huge resultset?).
We can add this workaround only for Vertica (as most of databases don't experience problems with SELECT ... LIMIT ... queries).

**Comment by hannahshachar** [comment]
But why should we have to disable this option?  If it worked before, why shouldn't it continue to work now?

**Comment by hannahshachar** [comment]
![image](https://user-images.githubusercontent.com/18379198/65154593-73f37380-da34-11e9-8b04-27df9c3f3495.png)
On my end the option is already disabled, but I have the same problem.  I have to open a different tool just to run commands like close_session that start will select but cannot have limit.

**Comment by Maxpowerfreak** [comment]
Hello, after upgrading from 6.1.5 to 6.2.1, I have recently encountered the same issue when trying to run functions on Vertica (in my case, `SELECT ANALYZE_STATISTICS('table_name');`). It looks like a recent change has started including an implicit limit X to the query.

I am running the most recent 8.X Vertica Driver (running an old version at work), so this should not be a driver related as my driver version differs from the OP.

Let me know if you need more information about my DBeaver setup!

**Comment by oBoMBaYo** [comment]
I found a workaround. You can just put `select 1;` right after your query and select the whole area then execute it. 
Example: 
`select analyze_statistics('schema_name.table_name'); select 1;`
![Screen Shot 2562-09-27 at 12 17 40](https://user-images.githubusercontent.com/10667447/65743931-d3324180-e120-11e9-9601-8822df101159.png)

**Comment by rudmer** [comment]
> I can't reproduce this.
> It fails with this error only if "Use SQL to limit fetch size" is enabled (DBeaver appends explicit LIMIT to all SELECT queries in this case).
> But with this option disabled everything works ok.
> 
> @rudmer Please check that this option is disabled for your particular Vertica connection:

@serge-rider 
This happens on Vertica 9.0 with the vertica 9.0 jdbc driver
The option is (and was) already disabled (see screenshot above)

**Comment by rudmer** [comment]
@serge-rider I just installed the latest version 6.2.1 and the problem was still there so I dug a little deeper and found the cause!

when going to version 6.something I was asked for conversion of my config (recommended) so I went ahead and said yes. After a couple of days things started to get weird since some settings were apparently not copied or applied and so I fixed those.

now today I found that while the general setting for the SQL data-editor `Use SQL to limit fetch size` = `false` the individual connection settings are `true`... :scream: this was never set by me and so I never checked...

So setting each connection not the use SQL for limits fixes this problem!
